sdk/src/main/java/co/omise/android/api/EndPoint.kt

```diff
@@ -0,0 +1,105 @@
+package co.omise.android.api
+
+import okhttp3.CertificatePinner
+import okhttp3.HttpUrl
+import java.util.*
+
+
+/**
+ * Endpoints encapsulates information about a particular Omise API endpoint.
+ * Currently there exists 2 endpoints, the API and the VAULT.
+ *
+ *
+ * This class encapsulates the following information for each endpoint:
+ *
+ *  * Host and network scheme (defaults to HTTPS.)
+ *  * The certificate hash to pin against.
+ *  * Whether to use the public key or the secret key.
+ *
+ */
+abstract class Endpoint {
+
+    /**
+     * The scheme to use, defaults to HTTPS.
+     *
+     * @return A [String] containing the network scheme to use.
+     */
+    private fun scheme(): String {
+        return "https"
+    }
+
+    /**
+     * The host name to connect to.
+     *
+     * @return A [String] containing the host name to connect to.
+     */
+    abstract fun host(): String
+
+    /**
+     * The certificate hash to use with OkHttp's [CertificatePinner].
+     * The default implementation returns a certificate hash for `*.omise.co` domains.
+     *
+     * @return A [String] containing the cert hash to pin against or `null` to
+     * pin no certificate.
+     */
+    fun certificateHash(): String {
+        return "sha256/maqNsxEnwszR+xCmoGUiV636PvSM5zvBIBuupBn9AB8="
+    }
+
+    /**
+     * The authentication key to use. The key should be taken from the given [Config] object.
+     * Either [Config.publicKey] or [Config.secretKey] should be returned.
+     *
+     * @param config A [Config] instance.
+     * @return A [String] containing the authentication key.
+     */
+    abstract fun authenticationKey(config: Config): String?
```

Comment:
I don't think it should return nullable data.

Reply:
Yeah leftover from having both keys, will change 👍

Reply:
👌